src/propagation/parameters.jl

```diff
@@ -0,0 +1,113 @@
+- `abstol::T`: absolute tolerance used to compute propagation timestep (default: `1e-20`).
+- `parse_eqs::Bool`: whether to use the specialized method of `jetcoeffs` or not (default: `true`).
+- `bwdoffset/fwdoffset::T`: days to propagate beyond first (bwd) / last (fwd) observation (default: `0.5`).
+- `coeffstol::T`: maximum size of the coefficients (default: `10.0`).
+
+# Residuals Parameters
+
+- `debias_table::String`: debiasing scheme (default: `"2018"`). Possible values are:
+    - `"2014"` corresponds to https://doi.org/10.1016/j.icarus.2014.07.033,
+    - `"2018"` corresponds to https://doi.org/10.1016/j.icarus.2019.113596,
+    - `"hires2018"` corresponds to https://doi.org/10.1016/j.icarus.2019.113596.
+
+# Least Squares Fit Parameters
+
+- `niter::Int`: number of iterations for differential corrections / Newton's method (default: `5`).
+
+# Gauss Method Parameters
+
+- `max_triplets::Int`: maximum number of triplets to check for a solution (default: `10`).
+- `varorder::Int`: order of jet transport perturbation (default: `5`).
+- `gaussQmax::T`: nrms threshold (default: `5.0`).
+
+# Admissible Region Parameters
+
+- `H_max::T`: maximum absolute magnitude (default: `32.0`).
+- `maxiter::Int`: maximum number of iterations for admissible region `ADAM` optimizer (default: `100`).
+- `tsaQmax::T`: nrms threshold (default: `1.5`).
+
+# Outlier Rejection Parameters
+
+- `max_per::T`: maximum allowed rejection percentage (default: `18.0`).
+"""
+function NEOParameters(; maxsteps::Int = 500, μ_ast::Vector{T} = μ_ast343_DE430[1:end],
+                      order::Int = 25, abstol::T = 1e-20, parse_eqs::Bool = true, bwdoffset::T = 0.5,
+                      fwdoffset::T = 0.5, coeffstol::T = 10.0, debias_table::String = "2018",
+                      niter::Int = 5, max_triplets::Int = 10, varorder::Int = 5, gaussQmax::T = 5.0,
+                      H_max::T = 32.0, maxiter::Int = 100, tsaQmax::T = 1.5, max_per::T = 18.0) where {T <: AbstractFloat}
+    # Unfold debiasing matrix
+    mpc_catalogue_codes_201X, truth, resol, bias_matrix = select_debiasing_table(debias_table)
+    # Sun (Earth) ephemeris
+    _su = selecteph(sseph, su)
+    eph_su = TaylorInterpolant(_su.t0, _su.t, collect(_su.x))
+    _ea = selecteph(sseph, ea)
+    eph_ea = TaylorInterpolant(_ea.t0, _ea.t, collect(_ea.x))
+    # Assemble NEOParameters
+    return NEOParameters{T}(maxsteps, μ_ast, order, abstol, parse_eqs, bwdoffset,
+                            fwdoffset, coeffstol, mpc_catalogue_codes_201X, truth,
+                            resol, bias_matrix, eph_su, eph_ea, niter, max_triplets,
+                            varorder, gaussQmax, H_max, maxiter, tsaQmax, max_per)
+end
+
+function NEOParameters(params::NEOParameters{T}; kwargs...) where {T <: AbstractFloat}
+    fields = fieldnames(NEOParameters{T})
+    vals = Vector{Any}(undef, length(fields))
+    for i in eachindex(vals)
+        if fields[i] in keys(kwargs)
+            vals[i] = kwargs[i]
```

Comment:
If I do
```julia
using NEOs
params = NEOParameters()
params = NEOParameters(params, parse_eqs=false)
```
then I get an error; one way to solve this is as follows:

```suggestion
            vals[i] = kwargs[fields[i]]
```

Reply:
This was fixed by https://github.com/PerezHz/NEOs.jl/pull/31/commits/e8bf9bb596236269685ab95f01e486a8ccf69003.